src/cluster_asm.c

```diff
@@ -1622,6 +1630,14 @@ void clusterSyncSlotsCommand(client *c) {
             return;
         }
 
+        /* Trimming is skipped on failover to avoid data loss with legacy
+         * migration. Failover during import may leave keys in unowned slots and
+         * once a replica becomes master we cannot tell if legacy method or ASM
+         * was used, since replicas are unaware of migrations. Admin may also
+         * mark unowned slots as migrating to continue the legacy migration.
+         * So, ASM defers trimming unowned slots until the next ASM operation.*/
+        asmTrimSlotsIfNotOwned();
```

Comment:
why do we call it here

Reply:
is it necessary?

Reply:
We don't have to but I think it is important to start trimming as soon as possible. I remember trimming was taking a very long time on some deployments (maybe with rof and modules). If we have keys in unowned slots, we should try to trim them before the next import operation. Otherwise, there is a risk to fail import if the same slots are being imported.